compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangErrorConstructorExpr.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.ballerinalang.compiler.tree.expressions;
+
+import org.ballerinalang.model.tree.NodeKind;
+import org.ballerinalang.model.tree.expressions.ErrorConstructorExpressionNode;
+import org.ballerinalang.model.tree.expressions.ExpressionNode;
+import org.ballerinalang.model.tree.expressions.NamedArgNode;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeVisitor;
+import org.wso2.ballerinalang.compiler.tree.types.BLangUserDefinedType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of error-constructor-expr.
+ *
+ * @since 2.0.0
+ */
+public class BLangErrorConstructorExpr extends BLangExpression implements ErrorConstructorExpressionNode {
+    public BLangUserDefinedType errorTypeRef;
+    public List<BLangExpression> positionalArgs = new ArrayList<>();
+    public List<BLangNamedArgsExpression> namedArgs = new ArrayList<>();
+    // This is added to store namedArgs and it is used only at desugar.
```

Comment:
Store the detail mapping created from the named args, right?

Reply:
Yes, fixed